### DISCUSSION TITLE
The logged-on user doesn't have admin rights

### DISCUSSION THREAD

**Discussion by m-spitfire** [farag2/Sophia-Script-for-Windows#212] [Q&A] [answered]
Hello, I am trying to run the script in the admin powershell but this warning comes and script exits:
```
WARNING: The logged-on user doesn't have admin rights
```
That's because it thinks that current user is not admin. It checks that with equality of `$CurrentUserName` and `$LoginUserName`:
```powershell
$CurrentUserName = (Get-Process -Id $PID -IncludeUserName).UserName | Split-Path -Leaf
$CurrentSessionId = (Get-Process -Id $PID -IncludeUserName).SessionId
$LoginUserName = (Get-Process -IncludeUserName -ErrorAction SilentlyContinue | Where-Object -FilterScript {($_.Name -eq "explorer") -and ($_.SessionId -eq $CurrentSessionId)}).UserName | Select-Object -Index 1 | Split-Path -Leaf
```
The `$CurrentUser` evaluates to `Murad` but `$LoginUserName` is just empty string. As you can see from the below picture that my user is Administrator:

![image](https://user-images.githubusercontent.com/66242799/125474398-66fec9fd-1d44-4e4f-80df-461810c76631.png)

Should I just delete the lines for checking if user is admin or not, or should I do something else?

**Comment by farag2** [comment]
Hi. Yes, you're right. I fixed it, re-released 2 hours later with the same version. Please, try again and let me know.

**Reply by farag2** [reply]
Also I see the edited line. When did you download the v5.11 version?

**Reply by farag2** [reply]
I tried the new one, and now it works for me.

**Reply by m-spitfire** [reply]
@farag2 I tried new one it doesn't work. The problem is not with the switch statement it's with the command for getting `$LoginUserName`, it sets it to the empty string. 

I just downloaded the v5.11 version but it was from a link from README, so it was an old version iirc

**Reply by farag2** [reply]
Hm... Do you a Telegram account? I want to see your error remotely. Via TeamViewer.

**Reply by m-spitfire** [reply]
Why are you checking for a process with the name "explorer"? I'm not familiar with powershell and windows in general but what if I don't have a explorer process?

**Reply by farag2** [reply]
So you work without a graphical user interface?) Explorer is the GUI for Windows.

**Reply by m-spitfire** [reply]
I do have telegram account but I won't show you via TeamViewer, I just don't feel comfortable + I don't want to install that proprietary program

I could make a screen recording of what's happening

**Reply by farag2** [reply]
OK. Waiting...

**Reply by m-spitfire** [reply]
> 
> 
> So you work without a graphical user interface?) Explorer is the GUI for Windows.

Ah I thought explorer was just the program name for "File explorer"

**Reply by m-spitfire** [reply]
Is there an easy way to record screen? All the sites say I should use a third party software and xbox gaming bar can't record powershell screen.

**Reply by General-neo** [reply]
> Hi. Yes, you're right. I fixed it, re-released 2 hours later with the same version. Please, try again and let me know.

I have been trying for last 1 hour thinking it's a problem at my end.  Its not working.

**Reply by farag2** [reply]
I know. Fixing...

**Reply by CyberWaspNinja** [reply]
@spitfire-hash - It is possible that you have previously uninstalled Powershell 2.0 and Engine. Do you want to  try adding them and the checking this out.

**Reply by farag2** [reply]
> @spitfire-hash - It is possible that you have previously uninstalled Powershell 2.0 and Engine. Do you want to try adding them and the checking this out.

It doesn't depend on this. PS 2.0 isn't being used in PS 5.1.

**Reply by CyberWaspNinja** [reply]
> > @spitfire-hash - It is possible that you have previously uninstalled Powershell 2.0 and Engine. Do you want to try adding them and the checking this out.
> 
> It doesn't depend on this. PS 2.0 isn't being used in PS 5.1.

Better to try and dismiss.. You can try from your end if @spitfire-hash is not readily available for testing. Remove 2.0 and Engine and try running script and see if this Error Appears. If it doesn't we can dismiss it. Unfortunatley, I am at work and I can't try

**Reply by farag2** [reply]
Tested on a clear 21H1 VM: disabled both, restarted. No error appears. :)

**Reply by CyberWaspNinja** [reply]
Funny - I had the same issue yesterday and I ran out of wits to fix this. I gave almost all SePrivilege and still no luck.. Finally, I turned on Powershell 2.0 and it worked :)

**Comment by farag2** [comment]
Why not to use a proprietary app?.. google for screen to gif

**Reply by m-spitfire** [reply]
> Why not to use a proprietary app?.. google for screen to gif

http://www.gnu.org/philosophy/free-sw.html
Btw I recorded it:

https://user-images.githubusercontent.com/66242799/125482428-c5efb24e-d4a5-4729-93c1-0821fc2bd519.mov

Edit: uploaded as mov

**Comment by syn990** [comment]
It would be possible to use OBS. instead of streaming you simply configure the screen as the source and send the output as a file.

https://obsproject.com/

Sent with [ProtonMail](https://protonmail.com/) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Reply by m-spitfire** [reply]
Yeah I knew that, but didn't wanted to install

Then couldn't find anything built in then installed it.

**Comment by farag2** [comment]
[Sophia Script v5.11.1.zip](https://github.com/farag2/Windows-10-Sophia-Script/files/6810281/Sophia.Script.v5.11.1.zip)
Check this. Hopefully, it works for us all.

**Reply by General-neo** [reply]
It seems to be now.  Let me complete the whole process

**Reply by General-neo** [reply]
It worked. Thank you.  Amazing

**Reply by farag2** [reply]
I'll release 5.11.1 soon

**Comment by farag2** [comment, marked as answer]
`5.11.1`
https://github.com/farag2/Windows-10-Sophia-Script/releases/latest
